libraries/AP_HAL_Linux/GPIO_RPI.cpp

```diff
@@ -29,7 +29,7 @@
 #define GPIO_MODE_ALT(g,a)  *(_gpio+(((g)/10))) |= (((a)<=3?(a)+4:(a)==4?3:2)<<(((g)%10)*3))
 #define GPIO_SET_HIGH       *(_gpio+7)  // sets   bits which are 1
 #define GPIO_SET_LOW        *(_gpio+10) // clears bits which are 1
-#define GPIO_GET(g)         (*(_gpio+13)&(1<<g)) // 0 if LOW, (1<<g) if HIGH
+#define GPIO_GET(g)         ((g >= 32) ? 0 : (*(_gpio+13)&(1<<g))) // 0 if LOW, (1<<g) if HIGH
```

Comment:
IMO the check should be on the caller, not here.

Reply:
Moved. I had stuck it in the macro to ensure any future users of the macro were protected, but it's currently only used in one location anyways so was easy to put in the caller.